slam-libraries/viam-orb-slam3/orbslam_server_v1.cc

```diff
@@ -778,13 +779,24 @@ double ReadTimeFromFilename(string filename) {
 
     // Now we read from buffer using get_time manipulator
     // and formatting the input appropriately.
-    ss >> std::get_time(&dt, "%Y-%m-%dT%H_%M_%SZ");
-    double sub_sec =
-        (double)std::stof(filename.substr(filename.find(".")), &sz);
+    ss >> std::get_time(&dt, time_format.c_str());
     time_t thisTime = std::mktime(&dt);
+    auto sub_sec_index = filename.find(".");
+    if ((sub_sec_index != string::npos)) {
```

Comment:
Why will there not always be sub-seconds?

Reply:
because the .osa files would save with .0000 before, the new time format in rdk would end up removing the sub second portion when loading from an apriori map. I can add a comment in the code to clarify this

Reply:
Does that mean this can go away? (Actually, do you think you could put TODOs over everything that will go away?)

Reply:
no this one would need to stay, unless we wanted to remove sub seconds from the yaml/osa files entirely

Reply:
Ah, thanks